provider/lxd/environ.go

```diff
@@ -470,31 +471,46 @@ func (env *environ) DetectSeries() (string, error) {
 
 // DetectHardware returns the hardware characteristics for the controller for
 // this environment. This method is part of the environs.HardwareCharacteristicsDetector
-// interface. On an LXD cloud, it must first check if it is a localhost cloud,
-// in that case, we only fill the Arch.
+// interface. On an LXD cloud, it must first check if it is a local cloud, in
+// that case, we only fill the Arch constraint.
+// This method should always return nil errors, because it is run during
+// bootstrap and its only purpose is to update the Arch constraint so in case
+// of an error the default arch is used and the bootstrap can continue.
 func (env *environ) DetectHardware() (*instance.HardwareCharacteristics, error) {
-	// The returned error is willingly ignored, because this should not
-	// break bootstrapping (we detect hardware before bootstrapping),
-	// instead, bootstrapping should fallback to default hardware arch.
-	endpointURL, _ := url.Parse(env.cloud.Endpoint)
+	// In order to determine if this is a local lxd cloud, we need to
+	// extract its endpoint IP.
+	// The endpoint can be formatted either as https://host:port,
+	// https://host, host:port, host, https://IP:port, https://IP,
+	// IP:port, or only IP, so we try to parse it as a URL but first
+	// ensuring it contains the correct scheme.
+	endpointURL, err := url.Parse(lxd.EnsureHTTPS(env.cloud.Endpoint))
+	if err != nil {
+		return nil, nil
```

Comment:
I would be inclined to log this error if we get one, but I suppose the bootstrap would fail later if we can't parse it here.

Reply:
yeah I agree, I added a debug log there

Reply:
I wouldn't put 'lxd' in here. "error parsing endpoint as URL: "
we are doing it because it *might* be an LXD endpoint, but I don't think we *know* at this point that it is LXD